File: go-sdk/pkg/models/operations/getfilesuploadsessionsidparts.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

var GetFilesUploadSessionsIDPartsServerList = []string{
	"https://upload.box.com/api/2.0",
}

type GetFilesUploadSessionsIDPartsPathParams struct {
	UploadSessionID string `pathParam:"style=simple,explode=false,name=upload_session_id"`
}

type GetFilesUploadSessionsIDPartsQueryParams struct {
	Limit  *int64 `queryParam:"style=form,explode=true,name=limit"`
	Offset *int64 `queryParam:"style=form,explode=true,name=offset"`
}

type GetFilesUploadSessionsIDPartsRequest struct {
	Retries     *utils.RetryConfig
	ServerURL   *string
	PathParams  GetFilesUploadSessionsIDPartsPathParams
	QueryParams GetFilesUploadSessionsIDPartsQueryParams
}

type GetFilesUploadSessionsIDPartsResponse struct {
	ClientError *shared.ClientError
	ContentType string
	StatusCode  int64
	UploadParts *shared.UploadParts
}

File: sdks/ts/src/sdk/models/operations/getfilesuploadsessionsidparts.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";

export const GetFilesUploadSessionsIdPartsServerList = [
	"https://upload.box.com/api/2.0",
] as const;


export class GetFilesUploadSessionsIdPartsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=upload_session_id" })
  uploadSessionId: string;
}


export class GetFilesUploadSessionsIdPartsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=offset" })
  offset?: number;
}


export class GetFilesUploadSessionsIdPartsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  serverURL?: string;

  @SpeakeasyMetadata()
  pathParams: GetFilesUploadSessionsIdPartsPathParams;

  @SpeakeasyMetadata()
  queryParams: GetFilesUploadSessionsIdPartsQueryParams;
}


export class GetFilesUploadSessionsIdPartsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  uploadParts?: shared.UploadParts;
}
